mage/ginkgo/lib.go

```diff
@@ -166,3 +178,111 @@ func (ns *NsWrapper) WaitForOutputContainsAndLog(description string, kcmd kubect
+	json := "jsonpath={.items}"
+	k := kubectl.Get("pods").
+		WithLabel(fmt.Sprintf("cassandra.datastax.com/datacenter=%s", dcName)).
+		FormatOutput(json)
+	ns.WaitForOutputAndLog(step, k, "[]", 300)
+}
+
+func (ns *NsWrapper) WaitForDatacenterOperatorProgress(dcName string, progressValue string, timeout int) {
+	step := fmt.Sprintf("checking the cassandra operator progress status is set to %s", progressValue)
+	json := "jsonpath={.status.cassandraOperatorProgress}"
+	k := kubectl.Get("CassandraDatacenter", dcName).
+		FormatOutput(json)
+	ns.WaitForOutputAndLog(step, k, progressValue, timeout)
+}
+
+func (ns *NsWrapper) WaitForDatacenterReadyPodCount(dcName string, count int) {
+	timeout := count * 400
+	step := "waiting for the node to become ready"
+	json := "jsonpath={.items[*].status.containerStatuses[0].ready}"
+	k := kubectl.Get("pods").
+		WithLabel(fmt.Sprintf("cassandra.datastax.com/datacenter=%s", dcName)).
+		WithFlag("field-selector", "status.phase=Running").
+		FormatOutput(json)
+	ns.WaitForOutputAndLog(step, k, duplicate("true", count), timeout)
+}
+
+func (ns *NsWrapper) WaitForDatacenterReady(dcName string) {
+	json := "jsonpath={.spec.size}"
+	k := kubectl.Get("CassandraDatacenter", dcName).FormatOutput(json)
+	sizeString := ns.OutputPanic(k)
+	size, err := strconv.Atoi(sizeString)
+	Expect(err).ToNot(HaveOccurred())
+
+	ns.WaitForDatacenterReadyPodCount(dcName, size)
+	ns.WaitForDatacenterOperatorProgress(dcName, "Ready", 30)
+}
+
+func (ns *NsWrapper) WaitForPodNotStarted(podName string) {
+	step := "verify that the pod is no longer marked as started"
+	k := kubectl.Get("pod").
+		WithFlag("field-selector", "metadata.name="+podName).
+		WithFlag("selector", "cassandra.datastax.com/node-state=Started")
+	ns.WaitForOutputAndLog(step, k, "", 60)
+}
+
+func (ns *NsWrapper) WaitForPodStarted(podName string) {
+	step := "verify that the pod is marked as started"
+	json := "jsonpath={.items[*].metadata.name}"
+	k := kubectl.Get("pod").
+		WithFlag("field-selector", "metadata.name="+podName).
+		WithFlag("selector", "cassandra.datastax.com/node-state=Started").
+		FormatOutput(json)
+	ns.WaitForOutputAndLog(step, k, podName, 60)
+}
+
+func (ns *NsWrapper) DisableGossipWaitNotReady(podName string) {
+	ns.DisableGossip(podName)
+	ns.WaitForPodNotStarted(podName)
+}
+
+func (ns *NsWrapper) EnableGossipWaitReady(podName string) {
+	ns.EnableGossip(podName)
+	ns.WaitForPodStarted(podName)
+}
+
+func (ns *NsWrapper) DisableGossip(podName string) {
+	execArgs := []string{"-c", "cassandra",
+		"--", "bash", "-c",
+		"nodetool disablegossip",
+	}
+	k := kubectl.ExecOnPod(podName, execArgs...)
+	ns.ExecVPanic(k)
+}
+
+func (ns *NsWrapper) EnableGossip(podName string) {
+	execArgs := []string{"-c", "cassandra",
+		"--", "bash", "-c",
+		"nodetool enablegossip",
+	}
+	k := kubectl.ExecOnPod(podName, execArgs...)
+	ns.ExecVPanic(k)
+}
+
+func (ns *NsWrapper) RetrieveDatacenterPodNames(dcName string) []string {
```

Comment:
I would go with
```suggestion
func (ns *NsWrapper) GetDatacenterPodNames(dcName string) []string {
```
to stick with the Kubernetes verbs